### DISCUSSION TITLE
"Global" Diagnostics and jumping to each error using the keyboard

### DISCUSSION THREAD

**Discussion by gak** [zed-industries/zed#9794] [Feature Requests]
I want to be able to jump to the first error of all errors in my project using the keyboard, (maybe) fix it, the press the keyboard shortcut to jump to the next error. This should continue until the last error on the list, then the next jump should be the first warning, and so on.

This is similar to "editor::GoToDiagnostic" but on the whole project and in order of "severity".

The closest I've found is "diagnostics: deploy" (https://github.com/zed-industries/zed/issues/297). It brings up rust errors and warnings (and presumably other language based diagnostics) into a new pane seemingly ordered by file. Can I jump to the next diagnostic from another pane? How can I jump to the file where the cursor is? This is a similar "pane: deploy search" action--can that jump to a file with a keyboard shortcut?

Also there is this discussion to hide warnings in diagnostics, but the changelog link is broken: https://github.com/zed-industries/zed/discussions/6652 Edit: Looks like it’s just in the config: {"diagnostics": { "include_warnings": false }}.

**Comment by alexpovel** [comment]
Second this, a "global" diagnostics, with configurable severity levels would be a welcome addition. The latter could work like either...

- one keyboard shortcut/command: cycle through errors if any, next cycle through warnings if any, ..., _or_
- multiple shortcuts/commands: have separate cycles for each severity level

In VSCode, there is [Go to next error](https://marketplace.visualstudio.com/items?itemName=yy0931.go-to-next-error) which enables this behavior. It works around VSCode not supporting this natively, cf. https://github.com/microsoft/vscode/issues/105795 .

**Comment by afiskon** [comment]
+1, that would be extremely helpful.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
